src/Fhir.Anonymizer.Core/PartitionedExecution/FhirStreamConsumer.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+
+namespace Fhir.Anonymizer.Core.PartitionedExecution
+{
+    public class FhirStreamConsumer : IFhirDataConsumer, IDisposable
+    {
+        private StreamWriter _writer;
+
+        public FhirStreamConsumer(Stream stream)
+        {
+            _writer = new StreamWriter(stream);
+        }
+
+        public async Task CompleteAsync()
+        {
+            await _writer.FlushAsync().ConfigureAwait(false);
+        }
+
+        public async Task ConsumeAsync(IEnumerable<string> data)
+        {
+            foreach (string content in data)
+            {
+                await _writer.WriteLineAsync(content).ConfigureAwait(false);
```

Comment:
What is the average size of content (resource) here? Does it make sense to use WriteAsync (for larger data size) here?